common/src/main/java/com/microsoft/identity/common/internal/ui/browser/BrowserDescriptor.java

```diff
@@ -0,0 +1,136 @@
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.google.gson.annotations.SerializedName;
+import com.microsoft.identity.common.internal.util.StringUtil;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Set;
+
+public class BrowserDescriptor implements Serializable {
+    @SerializedName("browser_package_name")
+    private String mPackageName;
+
+    @SerializedName("browser_signature_hashes")
+    private Set<String> mSignatureHashes;
+
+    @SerializedName("browser_use_customTab")
+    private boolean mUseCustomTab;
+
+    @SerializedName("browser_version_lower_bound")
+    private String mVersionLowerBound;
+
+    @SerializedName("browser_version_upper_bound")
+    private String mVersionUpperBound;
+
+    public BrowserDescriptor(
+            @NonNull final String packageName,
+            @NonNull final Set<String> signatureHashes,
+            final boolean useCustomTab,
+            @Nullable final String versionLowerBound,
+            @Nullable final String versionUpperBound) {
+        mPackageName = packageName;
+        mSignatureHashes = signatureHashes;
+        mUseCustomTab = useCustomTab;
+        mVersionLowerBound = versionLowerBound;
+        mVersionUpperBound = versionUpperBound;
+    }
+
+    public BrowserDescriptor(
+            @NonNull final String packageName,
+            @NonNull final String signatureHash,
+            final boolean useCustomTab,
+            @Nullable final String versionLowerBound,
+            @Nullable final String versionUpperBound) {
+        mPackageName = packageName;
+        mSignatureHashes = Collections.singleton(signatureHash);
+        mUseCustomTab = useCustomTab;
+        mVersionLowerBound = versionLowerBound;
+        mVersionUpperBound = versionUpperBound;
+    }
+
+    public boolean matches(@NonNull Browser browser) {
+        if (!mPackageName.equalsIgnoreCase(browser.getPackageName())) {
+            return false;
+        }
+
+        if (!mSignatureHashes.equals(browser.getSignatureHashes())) {
+            return false;
+        }
+
+        if (mUseCustomTab != browser.isCustomTabsServiceSupported()) {
+            return false;
+        }
+
+        if (!StringUtil.isEmpty(mVersionLowerBound)
+                && compareSemanticVersion(browser.getVersion(), mVersionLowerBound) == -1) {
+            return false;
+        }
+
+        if (!StringUtil.isEmpty(mVersionUpperBound)
+                && compareSemanticVersion(browser.getVersion(), mVersionUpperBound) == 1) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * The function to compare the two versions.
+     *
+     * @param thisVersion
+     * @param thatVersion
+     * @return int -1 if thisVersion is smaller than thatVersion,
+     *         1 if thisVersion is larger than thatVersion,
+     *         0 if thisVersion is equal to thatVersion.
+     */
+    private int compareSemanticVersion(
```

Comment:
Didn't we just add this function somewhere else? For broker `hello()`?

Reply:
`BrokerUtils` -- this should be refactored into `common`, not copy/pasted

Reply:
Thank you, Brian. Addressed.